torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -692,11 +692,9 @@ void ShardingUtil::PrepareOutputShardingPropagation(
     // Register the sharded data placeholder to the tensor and its node.
     (*data_placeholders)[i] = sharded_data_placeholder;
     xtensor->data()->handle = (*data_placeholders)[i];
```

Comment:
So we were setting the device data and the device data IR's handle, instead of resetting the IR.

Reply:
yea, we should just delete that device data. So it can be created again correctly later when used.